audb/core/info.py

```diff
@@ -581,3 +633,66 @@ def usage(
     """
     db = header(name, version=version, cache_root=cache_root)
     return db.usage
+
+
+def _filter_dependencies(
```

Comment:
The problem is that in `audb.load()` we also need access to `requested_tables()` and `requested_media()` after calling `audb.dependencies()`. This would not make much sense if we move the filtering to take place inside `audb.dependencies()`.

As an alternative we could add `tables` and `media` arguments to `audb.dependencies()`, but not hand them over inside `audb.load()`. This would look strange, but would be a possibility.

Reply:
> The problem is that in audb.load() we also need access to requested_tables() and requested_media() after calling audb.dependencies()

Can't we add `Dependencies.requested_tables` and `Dependencies.requested_media`?

Reply:
Or maybe simply use `Dependencies.tables` and `Dependencies.media`?

Reply:
I moved the filtering now inside `audb.dependencies()`. This involved quite a few changes, e.g. we now copy only files from archives that are requested as other files are no longer inside `deps()`.
The advantage is that we now use the same filter behavior in all our functions.

If you think the implementation is useful I would propose to first create another pull request that implements the changes to `audb.load()`.

Another downside is that we are now slower when using `audb.load(..., tables=[...])` as we first have to download the table with `audb.load_table()` to filter the dependencies and later on load them again into the flavor folder. As I see it this does not re-use the table from the main db cache. I guess we should be able to fix this.
Another related problem is that we then might need to revisit the progress bar we show when loading the tables, as now they are first loaded for deps (without progress bar at the moment).

Reply:
Ah ok, my idea was actually that we make `tables` and `media` arguments of the `Dependencies` class, that's why I proposed to then rely on `Dependencies.tables` and `Dependencies.media`. But not sure if this is a good idea / practical.

Reply:
We rely now on `Dependencies.table_ids` and `Dependencies.media`, compare https://github.com/audeering/audb/blob/a63addcea4d9981a8cfa35d7584023be760b32d6/audb/core/load.py#L973-L974